Added tests for temp table index rollback semantics fix

ayushdsh opened the pull request:
Temp table indexes starting without a '#' were not marked and hence did not follow rollback semantics leading to crashes and "cache lookup" failures

Engine PR: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/626

### Issues Resolved

Task: BABEL-6096, BABEL-5982

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

ayushdsh committed:
Added tests for temp table index rollback semantics fix

Temp table indexes starting without a '#' were not marked and hence did
not follow rollback semantics leading to crashes and "cache lookup"
failures

Signed-off-by: Ayush Shah <ayushdsh@amazon.com>

lejaokri commented on test/JDBC/expected/temp_table_rollback.out:
```diff
@@ -47,3 +47,252 @@ text
+
+SELECT * FROM #temp_with_indexes;
+GO
+~~START~~
+varchar#!#int
+a#!#1
+~~END~~
```
the transaction rolled back. Shouldn't this be empty?

ayushdsh replied:
tsql does not behave like postgres unless XACT_ABORT = ON. so only the second insert fails, first succeeds and since we are committing the transaction, we'll see 1 tuple.

i will also add a test with XACT_ABORT = ON.

lejaokri commented on test/JDBC/expected/temp_table_rollback.out:
```diff
@@ -47,3 +47,252 @@ text
+    CREATE TABLE #t1_proc1(a int)
+    INSERT INTO #t1_proc1 values (6)
+    BEGIN TRAN;
+        ALTER TABLE #t1_proc1 ADD b varchar(50)
```
nice! 

can we verify the enr-list after this alter command? I would expect a pg_toast table is created properly with the correct name starting with '#'

lejaokri replied:
While at it, add alter table drop column as well

ayushdsh committed:
Added tests for temp table index rollback semantics fix

Temp table indexes starting without a '#' were not marked and hence did
not follow rollback semantics leading to crashes and "cache lookup"
failures

Signed-off-by: Ayush Shah <ayushdsh@amazon.com>

ayushdsh committed:
Added tests for temp table index rollback semantics fix

Temp table indexes starting without a '#' were not marked and hence did
not follow rollback semantics leading to crashes and "cache lookup"
failures

Signed-off-by: Ayush Shah <ayushdsh@amazon.com>

ayushdsh committed:
Added tests for temp table index rollback semantics fix

Temp table indexes starting without a '#' were not marked and hence did
not follow rollback semantics leading to crashes and "cache lookup"
failures

Signed-off-by: Ayush Shah <ayushdsh@amazon.com>

ayushdsh committed:
Added tests for temp table index rollback semantics fix

Temp table indexes starting without a '#' were not marked and hence did
not follow rollback semantics leading to crashes and "cache lookup"
failures

Signed-off-by: Ayush Shah <ayushdsh@amazon.com>

ayushdsh committed:
Added tests for temp table index rollback semantics fix

Temp table indexes starting without a '#' were not marked and hence did
not follow rollback semantics leading to crashes and "cache lookup"
failures

Signed-off-by: Ayush Shah <ayushdsh@amazon.com>

jsudrik reviewed (approved):
Approved.
